### PR TITLE
Add GitHub ISSUE_TEMPLATE.md, GH #808

### LINKED ISSUE #808: Project should use a GitHub issue template
To help contributors understand and assist users, the project should use an issue template. This page explains how to set up an templates for GitHub issues and pull requests:

https://github.com/blog/2111-issue-and-pull-request-templates

### PR DESCRIPTION
**Pull request by stuartellis**
This adds a GitHub issue template.

Closes #808.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+## Reproducing the behavior
+
+### Computer that is creating the backups
+
+* Operating system, e.g. `Ubuntu Server 14.04.5`:
+* Ruby version, e.g. `2.3.1`:
+* Version of backup gem, e.g. `4.3.0`:
+* Versions of tools or services, e.g. `MySQL 5.7.9`:
+
+### Computer that is restoring the backups
```
Comment (tombruijn) [inline .github/ISSUE_TEMPLATE.md]: if it's different

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+## Expected behavior
```
Comment (tombruijn) [inline .github/ISSUE_TEMPLATE.md]: We could use this format for it this step. Make sure people describe the behavior in steps

- When I perform a backup
- I expect it to backup

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## Actual behavior
```
Comment (tombruijn) [inline .github/ISSUE_TEMPLATE.md]: We could use this format for it this step:

- When I perform a backup
- it throw the following error:

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+## Reproducing the behavior
+
+### Computer that is creating the backups
+
+* Operating system, e.g. `Ubuntu Server 14.04.5`:
+* Ruby version, e.g. `2.3.1`:
+* Version of backup gem, e.g. `4.3.0`:
+* Versions of tools or services, e.g. `MySQL 5.7.9`:
```
Comment (tombruijn) [inline .github/ISSUE_TEMPLATE.md]: or -> and

File: .github/ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+* Operating system, e.g. `Ubuntu Server 14.04.5`:
+* Ruby version, e.g. `2.3.1`:
+* Version of backup gem, e.g. `4.3.0`:
+* Versions of tools or services, e.g. `OpenSSL 0.9.8zh`:
+
+### Steps to reproduce the behavior
+
+1.
+2.
+3.
```
Comment (tombruijn) [inline .github/ISSUE_TEMPLATE.md]: Maybe we can prefill a scenario?

1. gem install backup
2. backup perform [OPTIONS]
3. check backups on s3

Comment (tombruijn) [inline .github/ISSUE_TEMPLATE.md]: Maybe we can ask people to provide their `config.rb` and related models as well.
We could even ask for a small project with a reproducible example.

File: .github/ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+## Expected behavior
+
+## Actual behavior
+
+## Reproducing the behavior
+
+### Computer that is creating the backups
+
+* Operating system, e.g. `Ubuntu Server 14.04.5`:
```
Comment (tombruijn) [inline .github/ISSUE_TEMPLATE.md]: Maybe this format is better?

`* Operating system: e.g. Ubuntu Server 14.04.5`

**Comment by stuartellis** [comment]
I've revised this based on your comments. It occurred to me that "what went wrong" and "steps to reproduce" are basically the same question, so I've merged them in this version.

**Pull request merged**